app/sources/HFTclTemplateController.m

```diff
@@ -65,8 +66,65 @@
     command_uint16_bits,
     command_uint32_bits,
     command_uint64_bits,
+    command_hf_min_version_required,
 };
 
+bool parseVersionString(const char* s, long* major, long* minor, long* patch) {
+    if (!s || !major || !minor || !patch) return false;
+
+    *major = *minor = *patch = 0;
```

Comment:
I'm hesitant on using C strings when we have Objective-C strings. I think this be simplified by using `componentsSeparatedByString:@"."] and getting the first/second/third components out as NSStrings, then calling `intValue` or `integerValue` to convert to a primitive.
Also can this be an method instead of a standalone function?

Reply:
Updated. The new method is a little more graceful with input (e.g., `3.a` will yield a minor value of `0`.) I'm not sure if this is necessarily a bad thing.

I also made it a method. (Originally, since the versioning logic didn't need access to anything within the controller, I thought it'd be better as a suite of free-standing functions.)